Civi/Api4/Generic/AbstractSaveAction.php

```diff
@@ -142,7 +142,13 @@ protected function matchExisting(&$record) {
       $where = [];
       foreach ($record as $key => $val) {
         if (isset($val) && in_array($key, $this->match, TRUE)) {
-          $where[] = [$key, '=', $val];
+          if (empty($val)) {
```

Comment:
`empty()` isn't the same as `=== ''`. I favor the latter because the number zero doesn't need this special treatment. But we should also match null so I think it should be
```suggestion
          if ($val === '' || is_null($val)) {
```